src/Microsoft.AspNetCore.SystemWebAdapters/SystemWebAdaptersConversionExtensions.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Web;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters;
+
+public static class SystemWebAdaptersConversionExtensions
+{
+    public static HttpRequest AsSystemWeb(this HttpRequestCore request)
+    {
+        ArgumentNullException.ThrowIfNull(request);
+
+        return request.HttpContext.AsSystemWeb().Request;
+    }
+
+    public static HttpResponse AsSystemWeb(this HttpResponseCore response)
+    {
+        ArgumentNullException.ThrowIfNull(response);
+
+        return response.HttpContext.AsSystemWeb().Response;
+    }
+
+    public static HttpResponseCore AsAspNetCore(this HttpResponse response)
+    {
+        ArgumentNullException.ThrowIfNull(response);
+
+        return response;
```

Comment:
This relies on the implicit converter? But going the other way the implicit converters rely on the extensions? Be consistent.

Reply:
It needs a private field to do the conversion, so I've made the implicit operators the source of truth.

Reply:
switched to an internal property